### PR TITLE
Added warning message while using the commands with ADO server

### PR DESCRIPTION
**Pull request by roshan-sy**
Please make sure the code is following contribution guidelines in CONTRIBUTING.md

 - [ ] : This PR has a corresponding issue open in the Repository.
 - [ ] : Approach is signed off on the issue.

Fixes: #1151 

To show the warning message code resolves the organization URL.
the resolved URL is then compared with the know patterns.

https://dev.azure.com/*
*.visualstudio.com

If URL does not matches with know patterns the warning message is shown

### CODE REVIEW

**Review by gauravsaralMs — COMMENTED** [review]
- will be good to have a better PR title and some more details around the logic in description
- will be good to add test cases
- can you share the experience how does this looks to the customer both
  - against hosted
  - against onprem
- we need review from @aaronhallberg on the messaging
  - I will suggest to not say that the support is in preview as we don't plan to make this support GA

**Comment by roshan-sy** [comment]
image

image

**Comment by roshan-sy** [comment]
> * will be good to have a better PR title and some more details around the logic in description
> * will be good to add test cases
> * can you share the experience how does this looks to the customer both
>   
>   * against hosted
>   * against onprem
> * we need review from @aaronhallberg on the messaging
>   
>   * I will suggest to not say that the support is in preview as we don't plan to make this support GA

Working on test cases and changing the message

**Comment by roshan-sy** [comment]
Done with test cases and message changes

Message: 
`Support for Azure DevOps Server, is under development`

Test cases
```
    def test_check_organization_with_dev_url(self):
        showWarning = check_organization(self._TEST_DEVOPS_ORGANIZATION)
        self.assertEqual(False, showWarning)

    def test_check_organization_with_visual_studio_url(self):
        showWarning = check_organization(self._TEST_VISUAL_STUDIO_ORGANIZATION)
        self.assertEqual(False, showWarning)

    def test_check_organization_with_ado_server_url(self):
        showWarning = check_organization(self._TEST_ADO_SERVER_ORGANIZATION)
        self.assertEqual(True, showWarning)
```

File: azure-devops/azext_devops/dev/common/services.py [hunk 1]
```diff
@@ -423,6 +426,10 @@ def get_project_id_from_name(organization, project):
         return team_project.id
     return project
 
+def check_organization(organization):
```
Comment (gauravsaralMs) [inline azure-devops/azext_devops/dev/common/services.py:429]: check_organization_in_azure

Comment (gauravsaralMs) [reply azure-devops/azext_devops/dev/common/services.py:429]: name

Comment (roshan-sy) [reply azure-devops/azext_devops/dev/common/services.py]: Method name changed

Comment (roshan-sy) [reply azure-devops/azext_devops/dev/common/services.py]: image

Done

File: azure-devops/azext_devops/dev/common/services.py [hunk 2]
```diff
@@ -355,6 +355,10 @@ def resolve_instance_project_and_repo(
         _raise_team_project_arg_error()
     if repo_required and repo is None:
         _raise_repo_requird_arg_error()
+
+    if not check_organization_in_azure(organization):
+        logger.warning("Support for Azure DevOps Server, is under development")
```
Comment (gauravsaralMs) [inline azure-devops/azext_devops/dev/common/services.py:360]: @aaronhallberg 
need your input on this string

Comment (aaronhallberg) [reply azure-devops/azext_devops/dev/common/services.py:360]: I think we should be more explicit here. Something like "_The Azure DevOps Extension for the Azure CLI does not support Azure DevOps Server._"

Comment (gauravsaralMs) [reply azure-devops/azext_devops/dev/common/services.py:360]: @roshan-sy : Let's take this change and we will be good to merge this PR

**Review by gauravsaralMs — APPROVED** [review]

File: azure-devops/azext_devops/dev/common/services.py (hunk 2 above)
Comment (roshan-sy) [reply azure-devops/azext_devops/dev/common/services.py]: Done

**Pull request merged**